src/api/test-controller/index.js

```diff
@@ -300,6 +306,19 @@ export default class TestController {
     _expect$ (actual) {
         const callsite = getCallsiteForMethod('expect');
 
+        const addAssertionWarning = (value, builderClassName, warningMessage, warningCallsite) => {
```

Comment:
I think we can simplify this code in the following way:
```js
_expect$ (actual) {
   //....
    if (isClientFunction(actual))
        this._addWarning(WARNING_MESSAGE.assertedClientFunctionInstance, callsite);

    if (isSelector(actual))
        this._addWarning(WARNING_MESSAGE.assertedSelectorInstance, callsite);
}

_addWarning (message, callsite = void 0) {
    const renderedCallsite = renderCallsiteSync(...);
    
    this.warningLog.addWarning(...)
}
```
`isSelector` and `isClientFunction` can be placed in an external module like `src/client-functions/types.ts`. You can declare them as type guards.